src/app/sign-up/sign-up.component.ts

```diff
@@ -0,0 +1,26 @@
+import {ChangeDetectionStrategy, Component, inject} from '@angular/core';
+import {FormBuilder, Validators} from "@angular/forms";
+import {passwordMatchValidator} from "./utils/sign-up.utils";
+
+@Component({
+  selector: 'app-sign-up',
+  templateUrl: './sign-up.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SignUpComponent {
+  hideEmail: boolean = true;
+  hideConfirmEmail: boolean = true;
+  private fb = inject(FormBuilder);
+  signUpForm = this.fb.group({
```

Comment:
I'd like to separate public, private and inject entities

Reply:
fixed